File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/SideDrawerCustomContentModalController.swift
//
//  SideDrawerCustomContentModalController.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

class SideDrawerCustomContentModalController: SideDrawerGettingStartedModalController, TKSideDrawerDelegate {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let sideDrawer = self.sideDrawer
        sideDrawer.delegate = self
        sideDrawer.content = self.setupSideDrawerContent()
        sideDrawer.style.headerHeight = 64
    }
    
    func setupSideDrawerContent() -> UIView {
        let sideDrawerContent = UIView()
        sideDrawerContent.backgroundColor = UIColor.clearColor()
        let imageView = UIImageView(image:UIImage(named: "logo"))
        imageView.frame = CGRectMake(sideDrawer!.width / 2 - imageView.frame.size.width / 2, self.view.frame.size.height / 2 - imageView.frame.size.height, imageView.frame.size.width, imageView.frame.size.height)
        sideDrawerContent.addSubview(imageView)
        
        return sideDrawerContent
    }
}

File: TelerikUIExamplesInCSharp/Examples/SideDrawer/CustomContentExample/SideDrawerCustomContentModalController.cs
using System;

using ObjCRuntime;
using Foundation;
using UIKit;
using CoreAnimation;
using CoreGraphics;

using TelerikUI;

namespace Examples
{
	public class SideDrawerCustomContentModalController : SideDrawerGettingStartedModalController
	{
		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();
			this.SideDrawer.Content = this.SetupContent ();
		}

		public UIView SetupContent()
		{
			UIView sideDrawerContent = new UIView ();
			sideDrawerContent.BackgroundColor = UIColor.Clear;
			UIImageView imageView = new UIImageView (new UIImage ("logo"));
			imageView.Frame = new CGRect ((float)this.SideDrawer.Width / 2.0f - imageView.Frame.Size.Width / 2.0f, this.View.Frame.Size.Height / 2.0f - imageView.Frame.Size.Height, 
				imageView.Frame.Size.Width, imageView.Frame.Size.Height);
			sideDrawerContent.AddSubview (imageView);

			return sideDrawerContent;
		}
	}
}

